kubernetes/check.go

```diff
@@ -0,0 +1,57 @@
+package kubernetes
+
+import (
+	apiv1beta1 "github.com/docker/cli/kubernetes/compose/v1beta1"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	apimachinerymetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+)
+
+// StackVersion represents the detected Compose Component on Kubernetes side.
+type StackVersion string
+
+const (
+	// StackAPIV1Beta1 is returned if it's the most recent version available.
+	StackAPIV1Beta1 = StackVersion("v1beta1")
+)
+
+// GetAPIVersion returns the most recent stack API installed.
+func GetAPIVersion(kubeConfig *restclient.Config) (StackVersion, error) {
```

Comment:
We may want to use `GetStackAPIVersion` here, as it's only the `Stack` api version we're getting here. We could have more api extension in the future, and `kubernetes.GetAPIVersion` kinda makes think it's the kubernetes api (server) version.

Reply:
ok